dojo/tools/dependency_check/parser.py

```diff
@@ -10,172 +11,127 @@
+            if maven_node:
+                maven_parts = maven_node.findtext(f'{namespace}name').split(':')
+                # logger.debug('maven_parts:' + str(maven_parts))
+                if len(maven_parts) == 3:
+                    component_name = maven_parts[0] + ':' + maven_parts[1]
+                    component_version = maven_parts[2]
                     return component_name, component_version
 
-        except:
-            logger.exception('error parsing component_name and component_version')
-            logger.debug('dependency: %s', ElementTree.tostring(dependency, encoding='utf8', method='xml'))
-
-        return component_name, component_version
+        # TODO what happens when there multiple evidencecollectednodes with product or version as type?
+        evidence_collected_node = dependency.find(namespace + 'evidenceCollected')
+        if evidence_collected_node:
+            # <evidenceCollected>
+            # <evidence type="product" confidence="HIGH">
+            #     <source>file</source>
+            #     <name>name</name>
+            #     <value>jquery</value>
+            # </evidence>
+            # <evidence type="version" confidence="HIGH">
+            #     <source>file</source>
+            #     <name>version</name>
+            #     <value>3.1.1</value>
+            # </evidence>'
+            # will find the first product and version node. if there are multiple it may not pick the best
+            # since 6.0.0 howoever it seems like there's always a packageurl above so not sure if we need the effort to
+            # implement more logic here
+            product_node = evidence_collected_node.find('.//' + namespace + 'evidence[@type="product"]')
+            if product_node:
+                component_name = product_node.findtext(f'{namespace}value')
+                version_node = evidence_collected_node.find('.//' + namespace + 'evidence[@type="version"]')
+                if version_node:
+                    component_version = version_node.findtext(f'{namespace}value')
+
+                return component_name, component_version
+
+        return None, None
 
     def get_finding_from_vulnerability(self, dependency, related_dependency, vulnerability, test, namespace):
         dependency_filename, dependency_filepath = self.get_filename_and_path_from_dependency(dependency, related_dependency, namespace)
         # logger.debug('dependency_filename: %s', dependency_filename)
 
-        tags = []
-
         if dependency_filename is None:
             return None
 
-        name = self.get_field_value(vulnerability, 'name', namespace)
-        cwes_node = vulnerability.find(namespace + 'cwes')
-        if cwes_node is not None:
-            cwe_field = self.get_field_value(cwes_node, 'cwe', namespace)
+        tags = []
+        name = vulnerability.findtext(f'{namespace}name')
+        cwes_node = vulnerability.find(f'{namespace}cwes')
```

Comment:
The variable `cwes_node` is unused, so this line can be deleted.

Reply:
we should use it to set the CWE to the first value.